AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2664,6 +2668,67 @@ - (void)_enqueueAsyncSizingWithSentinel:(ASSentinel *)sentinel completion:(void(
   return self;
 }
 
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentCollection *)environmentCollection
+{
+  return &_environmentCollection;
+}
+
+- (ASDisplayNode *)parent
+{
+  return self.supernode;
+}
+
+- (void)setParent:(ASDisplayNode *)parent
```

Comment:
I don't think we should support a setParent: method in ASEnvironment.  Will it still work if the protocol only guarantees that an accessor method exists?